src/tokens.ts

```diff
@@ -278,6 +279,19 @@ export interface IGitExtension extends IDisposable {
    * @param path Path from which the top Git repository needs to be found
    */
   showTopLevel(path: string): Promise<Git.IShowTopLevelResult>;
+
+  /**
+   * Ensure a.gitignore file
+   *
+   */
+  ensureGitignore(): Promise<Response>;
+
+  /**
+   * Add an entry in .gitignore file
+   *
+   * @param filename The name of the entry to ignore
```

Comment:
```suggestion
   * @param filename The name of the entry to ignore
   * @param useExtension Ignore all file with the same extension than filename
```

Reply:
Thx for the review. Last commit implements this.